<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1420</width>
    <height>875</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1420</width>
    <height>875</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>InputHandler CLI generation tool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>1420</width>
     <height>850</height>
    </size>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1420</width>
      <height>850</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>1420</width>
      <height>850</height>
     </size>
    </property>
    <property name="sizeIncrement">
     <size>
      <width>0</width>
      <height>0</height>
     </size>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="ih_settings_tab">
     <attribute name="title">
      <string>InputHandler Settings</string>
     </attribute>
     <widget class="QPushButton" name="editButton_1">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>790</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Edit</string>
      </property>
     </widget>
     <widget class="QPushButton" name="defaultButton_1">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>790</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Default</string>
      </property>
     </widget>
     <widget class="QPushButton" name="clearButton_1">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>790</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Clear</string>
      </property>
     </widget>
     <widget class="QLabel" name="tab_1_code_preview_label">
      <property name="geometry">
       <rect>
        <x>1070</x>
        <y>0</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Code Preview</string>
      </property>
     </widget>
     <widget class="QTreeView" name="codePreview_1">
      <property name="geometry">
       <rect>
        <x>790</x>
        <y>20</y>
        <width>611</width>
        <height>801</height>
       </rect>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustToContents</enum>
      </property>
     </widget>
     <widget class="QLabel" name="tab_1_settings_table_label">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>0</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Settings</string>
      </property>
     </widget>
     <widget class="QTreeWidget" name="settings_tree">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>771</width>
        <height>761</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustToContents</enum>
      </property>
      <column>
       <property name="text">
        <string notr="true">1</string>
       </property>
      </column>
     </widget>
    </widget>
    <widget class="QWidget" name="command_tab">
     <attribute name="title">
      <string>Command Tree</string>
     </attribute>
     <widget class="QTreeWidget" name="command_tree">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>771</width>
        <height>631</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustToContents</enum>
      </property>
      <column>
       <property name="text">
        <string notr="true">1</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="tab_2_code_preview_label">
      <property name="geometry">
       <rect>
        <x>1070</x>
        <y>0</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Code Preview</string>
      </property>
     </widget>
     <widget class="QTreeView" name="codePreview_2">
      <property name="geometry">
       <rect>
        <x>790</x>
        <y>20</y>
        <width>611</width>
        <height>801</height>
       </rect>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustToContents</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="deleteButton_2">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>790</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Delete</string>
      </property>
     </widget>
     <widget class="QPushButton" name="editButton_2">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>790</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Edit</string>
      </property>
     </widget>
     <widget class="QPushButton" name="newButton_2">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>790</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>New</string>
      </property>
     </widget>
     <widget class="QLabel" name="tab_2_command_tree_label">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>0</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Commands</string>
      </property>
     </widget>
     <widget class="QWidget" name="command_settings" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>660</y>
        <width>781</width>
        <height>131</height>
       </rect>
      </property>
      <widget class="QTextEdit" name="textEdit">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>20</y>
         <width>381</width>
         <height>101</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_4">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>80</y>
         <width>41</width>
         <height>21</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="commandMinArgs">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>60</y>
         <width>41</width>
         <height>21</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="commandDepth">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>0</y>
         <width>41</width>
         <height>21</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox">
       <property name="geometry">
        <rect>
         <x>350</x>
         <y>20</y>
         <width>41</width>
         <height>21</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="commandParentId">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>60</y>
         <width>113</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="commandString">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>20</y>
         <width>113</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="commandLengthLabel">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>40</y>
         <width>47</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="functionName">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>0</y>
         <width>113</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>100</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Contains Wildcards</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>0</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Function name</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>0</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Command depth</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Command string</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Command length</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>47</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Parent id</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="commandId">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>80</y>
         <width>113</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>47</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>id</string>
       </property>
      </widget>
      <widget class="QLabel" name="commandSubcommands">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>20</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Subcommands</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>60</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Min. args.</string>
       </property>
      </widget>
      <widget class="QLabel" name="commandMaxArgs">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>80</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Max args.</string>
       </property>
      </widget>
      <widget class="QComboBox" name="commandArgumentHandling">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>40</y>
         <width>171</width>
         <height>20</height>
        </rect>
       </property>
       <property name="currentText">
        <string>No arguments</string>
       </property>
       <item>
        <property name="text">
         <string>No arguments</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Single argument type</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Argument type array</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>500</x>
         <y>0</y>
         <width>211</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Comma separated argument types</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1420</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="actionPreferences"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionInputHandler_Documentation"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuGenerate">
    <property name="title">
     <string>Generate</string>
    </property>
    <addaction name="actionGenerate_CLI_Files"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuGenerate"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionInputHandler_Documentation">
   <property name="text">
    <string>InputHandler Documentation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionGenerate_CLI_Files">
   <property name="text">
    <string>Generate CLI Files</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
